Lib/test/test_os.py

```diff
@@ -835,6 +835,31 @@ def test_key_type(self):
         self.assertIs(cm.exception.args[0], missing)
         self.assertTrue(cm.exception.__suppress_context__)
 
+    def _test_environ_iteration(self, iterator):
+        new_key = "__new_key__"
+
+        next(iterator)  # start iteration over os.environ.items
+
+        # add a new key in os.environ mapping
+        os.environ[new_key] = "test_environ_iteration"
+
+        next(iterator)  # force iteration over modified mapping
+        self.assertEqual(os.environ[new_key], "test_environ_iteration")
+
+        del os.environ[new_key]
```

Comment:
The new key should be removed even if the test is failed, otherwise it could affect other tests. Use try...finally.